### PR TITLE
remove guest when visitor login

### PR DESCRIPTION
**Pull request by versoul**

### CODE REVIEW

**Comment by artwells** [comment]
Thanks!

File: accounts-guest-server.js [hunk 1]
```diff
@@ -10,6 +10,24 @@ Accounts.removeOldGuests = function (before) {
     return res;
 };
 
+/**
+ *  drop guest when visitor logs in
+ *
+ */
+GuestUsers = new Mongo.Collection('guestUsers');
+Accounts.onLogin(function(par){
+    if(par.user.username.indexOf('guest') !== -1){
```
Comment (Zodiase) [inline accounts-guest-server.js]: User with username "Im_not_a_guest" would be considered a guest in this case?

Comment (artwells) [inline accounts-guest-server.js]: good catch.  could you file a bug?

File: accounts-guest-server.js [hunk 2]
```diff
@@ -10,6 +10,24 @@ Accounts.removeOldGuests = function (before) {
...
+        if(!GuestUsers.findOne({connection_id: par.connection.id})){
+            GuestUsers.insert({connection_id: par.connection.id, user_id: par.user._id});
+        }
+    }
+    else if(par.type !== 'resume'){
```
Comment (Zodiase) [inline accounts-guest-server.js]: I'm not familiar with `par.type` and I'd like to know in what case would it be 'resume'? The official doc only says 'The service name, such as "password" or "twitter".'. What are the possible values?

**Comment by artwells** [comment]
@versoul could you respond?

**Comment by versoul** [comment]
as I could notice 'resume' is shown after I refresh the page, and options it seems ('password', 'twitter' etc) after an entrance or an exit in the corresponding way

if username is "Im_not_a_guest" its bug =(

**Pull request merged**